src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -138,21 +139,93 @@ impl Default for WorkunitMetadata {
   }
 }
 
-#[derive(Clone, Default)]
+type CompletedWorkunitReceiver =
+  Arc<Mutex<Receiver<(SpanId, Option<WorkunitMetadata>, SystemTime)>>>;
+type CompletedWorkunitSender = Arc<Mutex<Sender<(SpanId, Option<WorkunitMetadata>, SystemTime)>>>;
+
+#[derive(Clone)]
 pub struct WorkunitStore {
   rendering_dynamic_ui: bool,
+  streaming_workunit_data: StreamingWorkunitData,
+  heavy_hitters_data: HeavyHittersData,
+}
+
+#[derive(Clone)]
+struct StreamingWorkunitData {
+  started_workunits_rx: Arc<Mutex<Receiver<Workunit>>>,
+  started_workunits_tx: Arc<Mutex<Sender<Workunit>>>,
+  completed_workunits_rx: CompletedWorkunitReceiver,
+  completed_workunits_tx: CompletedWorkunitSender,
+  workunit_records: Arc<Mutex<HashMap<SpanId, Workunit>>>,
+}
+
+impl StreamingWorkunitData {
+  fn new() -> StreamingWorkunitData {
+    let (started_workunits_tx, started_workunits_rx) = channel();
+    let (completed_workunits_tx, completed_workunits_rx) = channel();
+    StreamingWorkunitData {
+      started_workunits_tx: Arc::new(Mutex::new(started_workunits_tx)),
+      started_workunits_rx: Arc::new(Mutex::new(started_workunits_rx)),
+      completed_workunits_tx: Arc::new(Mutex::new(completed_workunits_tx)),
+      completed_workunits_rx: Arc::new(Mutex::new(completed_workunits_rx)),
+      workunit_records: Arc::new(Mutex::new(HashMap::new())),
+    }
+  }
+}
+
+#[derive(Clone)]
+struct HeavyHittersData {
   inner: Arc<Mutex<WorkUnitInnerStore>>,
```

Comment:
It looks like `WorkUnitInnerStore` is now only used to compute the "heavy hitters" for the UI, which is really awesome. I think that you could maybe rename it to make that clearer?